docs/recipe_overview.md

```diff
@@ -0,0 +1,436 @@
+1. Groups: what should be included/excluded from a Backup/Restore
+1. Hooks: actions to run on Pods at the required place in the Workflow
+
+Note that a Recipe resides in the same Namespace as the Application it protects,
+like a VRG. If the Application to protect spans multiple Namespaces, then use
+one VRG and one Recipe per Namespace.
+
+## Example Recipe
+
+Full example files of the [VRG](../hack/recipe_e2e/protect/vrg_busybox_primary.yaml)
+and [Recipe](../hack/recipe_e2e/protect/recipe_busybox.yaml) linked.
+
+### Basic components
+
+```yaml
+apiVersion: ramendr.openshift.io/v1alpha1
+kind: Recipe
+metadata:
+  name: recipe-busybox
+  namespace: recipe-test
+spec:
+  appType: busybox
+  hooks:
+    # content omitted
+  groups:
+    # content omitted
+  workflows:
+    # content omitted
+```
+
+### Workflows
+
+Workflows are the high level components desired for DR.
+These components are referenced in the VRG.
+
+Workflows have a unique name, and a sequence composed of a Hooks and Groups,
+whose details are defined in the Hooks and Groups sections.
+
+There are two Workflows that exist by default: CaptureWorkflow and RecoverWorkflow.
+
+```yaml
+# metadata omitted for example
+# groups, hooks omitted for example
+spec:
+  captureWorkflow:
+    sequence:
+    - hook: service-hooks/pre-backup
+    - group: instance-resources
+  recoverWorkflow:
+    sequence:
+    - group: pvcs
+    - hook: service-hooks/pre-restore
+    - group: deployments
+```
+
+Note that the sample above matches the Busybox sample requirements, but also adds
+Hooks for demonstration purposes.
+
+#### Workflow Sequencing
+
+Given the sample Workflow above:
+
+* The Capture sequence is: Hook -> Group. The Hook must complete before Group begins.
+  The Capture sequence is complete when Group finishes successfully.
+* The Recover sequence is: Group1 -> Hook -> Group2. Group1 must complete before
+  Hook runs, Hook must complete before Group2 begins, and Group2 begins after Hook
+  completes. The Recover sequence is complete when Group2 finishes successfully.
+
+#### Sample VRG connection
+
+The sample Recipe in this example is named `recipe-busybox`, and is in the same
+Namespace as the VRG shown below. To use this Recipe, the VRG needs to add its
+name to the `spec.kubeObjectProtection.recipeRef.name` field.
+
+```yaml
+# VRG metadata omitted
+spec:
+  kubeObjectProtection:
+    recipeRef:
+      name: recipe-busybox
+```
+
+### Groups
+
+#### Resources
+
+Groups define what should be included/excluded from a Backup or Restore.
+Since all resources needed can be backed up at the same time, this is defined
+in the group `instance-resources`. Since PVCs must be restored before Deployments,
+these are broken into two separate groups: `deployments` and `pvcs`, each limiting
+the resources to restore to the appropriate type.
+
+```yaml
+# metadata omitted
+spec:
+  groups:
+    - name: instance-resources
+      backupRef: instance-resources
```

Comment:
Since this is a _restore_ group I think the `backupRef` should be omitted for clarity.  This option would be removed if [capture and recover groups were defined](https://github.com/RamenDR/recipe/issues/10#issuecomment-1533797880).